include/mbedtls/config.h

```diff
@@ -1539,7 +1539,8 @@
 /**
  * \def MBEDTLS_SSL_FALLBACK_SCSV
  *
- * Enable support for FALLBACK_SCSV (draft-ietf-tls-downgrade-scsv-00).
+ * Enable support for RFC 7507: Fallback Signaling Cipher Suite Value (SCSV)
```

Comment:
I checked the [differences between the draft and the RFC](https://tools.ietf.org/rfcdiff?url1=https://tools.ietf.org/id/draft-ietf-tls-downgrade-scsv-00.txt&url2=https://tools.ietf.org/rfc/rfc7507.txt) and found the following potential differences:

* :negative_squared_cross_mark: A client must not include TLS_FALLBACK_SCSV if it sets its client version to its highest supported version. But we do include it even in that case (and we test that we do so).

Reply:
I'm not sure I get what you mean here. We're only sending that SCSV if the application explicitly instructed us do to so by calling `mbedtls_ssl_conf_fallback()`, whose documentation contains the following:
```
 * \warning        You should NOT set this to MBEDTLS_SSL_IS_FALLBACK for
 *                 non-fallback connections! This would appear to work for a
 *                 while, then cause failures when the server is upgraded to
 *                 support a newer TLS version.
```
Granted, we could probably replace "should" with "must" here, but we're already shouting "NOT" at our users, so hopefully that's explicit enough.

Or am I missing something?

Reply:
I expected the library to enforce this, rather than the application. Or `MBEDTLS_SSL_IS_FALLBACK` to mean “include `TLS_FALLBACK_SCSV` if this is a fallback” rather than “include `TLS_FALLBACK_SCSV` (and I assert that this is a fallback)”. But upon reflection the conditional meaning isn't really useful since the application has to use a different configuration for the fallback connection, not only with a different `fallback` value but also with a different `min_version`. And I guess we can't enforce that `fallback` isn't set when it shouldn't be since we don't know what `min_version` is for a non-fallback connection.

Upon reflection I don't see how we could improve the library behavior here. But I think we could improve the documentation. We're missing a high-level description of how to do version fallback securely, with two conf objects.

Reply:
Yes, only the application can know if this is a fallback, and we can't double-check.

But I agree we could improve the documentation.